### PR TITLE
EN-8343-optimize-trie-sync

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Instead of keeping the whole trie in memory during the syncing process, and Commit() the trie when it is fully synced, only keep the collapsed root and commit each node individually as it is synced.

### CODE REVIEW

File: data/trie/sync.go [hunk 1]
```diff
@@ -100,11 +100,6 @@ func (ts *trieSyncer) StartSyncing(rootHash []byte, ctx context.Context) error {
 
 		numUnResolved := ts.requestNodes()
 		if !shouldRetryAfterRequest && numUnResolved == 0 {
-			err = ts.trie.Commit()
```
Comment (sasurobert) [inline data/trie/sync.go:103]: why delete this ? isn't it better to have the final commit done ?

File: data/trie/sync.go [hunk 2]
```diff
@@ -186,6 +176,22 @@ func (ts *trieSyncer) checkIfSynced() (bool, error) {
...
+
+			if !ts.rootFound && bytes.Equal([]byte(nodeHash), ts.rootHash) {
+				var collapsedRoot node
+				collapsedRoot, err = currentNode.getCollapsed()
+				if err != nil {
+					return false, err
+				}
+
+				ts.trie.root = collapsedRoot
+				ts.rootFound = true
```
Comment (sasurobert) [inline data/trie/sync.go:193]: so we set root only after finding all its childs ?

File: data/trie/sync.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply data/trie/sync.go:103]: If we commit here, the whole trie will be loaded again from storage. The current approach is to commit each node as it is synced.

File: data/trie/sync.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply data/trie/sync.go:193]: Yes, it should have no impact upon the functioning of the algorithm.

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**